sql/parser/builtins.go

```diff
@@ -943,6 +961,89 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+func regexpReplace(s, pattern, to, flags string) (Datum, error) {
+	pattern, global := regexpEvalFlags(pattern, flags)
+
+	patternRe, err := regexp.Compile(pattern)
+	if err != nil {
+		return DNull, err
+	}
+
+	matchCount := 1
+	if global {
+		matchCount = -1
+	}
+
+	replaceIndex := 0
+	var newString bytes.Buffer
+	for _, matchIndex := range patternRe.FindAllStringSubmatchIndex(s, matchCount) {
+		start := matchIndex[0]
+		end := matchIndex[1]
+
+		preMatch := s[replaceIndex:start]
+		newString.WriteString(preMatch)
+
+		matchTo := to
+		match := s[start:end]
+		matchTo = strings.Replace(matchTo, "\\&", match, -1)
+		for sub := 1; 2*sub < len(matchIndex); sub++ {
+			subStart := matchIndex[2*sub]
+			subEnd := matchIndex[2*sub+1]
+
+			subMatch := s[subStart:subEnd]
+			matchTo = strings.Replace(matchTo, fmt.Sprintf("\\%d", sub), subMatch, -1)
+		}
+		newString.WriteString(matchTo)
+
+		replaceIndex = end
+	}
+	newString.WriteString(s[replaceIndex:])
+
+	return DString(newString.String()), nil
+}
+
+// regexpEvalFlags evaluates the provided Postgres regexp flags in
+// accordance with their definitions provided at
+// http://www.postgresql.org/docs/9.0/static/functions-matching.html#POSIX-EMBEDDED-OPTIONS-TABLE.
+// It then returns an adjusted regexp pattern, and a flag specifying if more
+// than one match should be found or not.
+func regexpEvalFlags(pattern, flags string) (string, bool) {
+	global := false
+	goReFlags := map[rune]bool{'s': true, 'm': true}
+
+	for _, flag := range flags {
+		switch flag {
+		case 'g':
+			global = true
+		case 'i':
+			goReFlags['i'] = true
+		case 'c':
+			delete(goReFlags, 'i')
+		case 's':
+			goReFlags['s'] = true
+		case 'm', 'n':
+			delete(goReFlags, 's')
+			delete(goReFlags, 'm')
```

Comment:
either way you're generating a string and then parsing it, so i think the approach i suggested is the way to go (if it works, i've never used `regexp/syntax`).